docs/developer-guide/extensions.md

```diff
@@ -0,0 +1,68 @@
+# Extensions
+
+An extension is a way to add new capabilities to the UI and API.
+
+Use cases include:
+
+* Surfacing high-level application telemetry and insights.
+* Recommending Kubernetes best practices.
+* Alerting users about vulnerabilities in their application.
+
+Extensions consist to one or two components:
+
+* A JavaScript module that exports UI extensions.
+* Optionally, an API extension that allows UI requests to be proxied to another service.
+
+Only install extensions from trusted sources.
+
+## UI Extension
+
+A Javascript module should be:
+
+* Packaged into a single `.js` file.
+* Install on each Argo CD server in `/tmp/extensions/{name}.js`
+* Listed in `/tmp/extensions/index.json`.
+
+You can add the following types of extension:
+
+* `appToolbarButton` - A button added to the application toolbar.
+* `appPanel` - A sliding panel.
+* `appStatusPanilItem` - An item added to the application status panel.
+* `resourcePanel` - An item added as a resource panel.
+
+Security:
+
+* An extension will be able to make HTTP requests.
```

Comment:
If the JS is loaded into the UI, doesn't it have the ability to do basically anything that's allowed by the page's security headers and the browser's default settings? This bullet point might give the impression that the plugin's behavior is more restricted than it actually is.